src/Transposit.ts

```diff
@@ -15,144 +15,120 @@
  */
 
 import { EndRequestLog } from ".";
-
-export interface ClientClaims {
-  iss: string; // issuer
-  sub: string; // subject
-  exp: number; // expiration
-  iat: number; // issuedAt
-  publicToken: string;
-  repository: string;
-  email: string;
-  name: string;
-}
-
-export interface OperationParameters {
-  [paramName: string]: string;
-}
-
-// https://my-app.transposit.io?clientJwt=...needKeys=... -> https://my-app.transposit.io
-function hereWithoutSearch(): string {
-  return `${window.location.origin}${window.location.pathname}`;
-}
-
-export const LOCAL_STORAGE_KEY = "TRANSPOSIT_SESSION";
+import { popCodeVerifier, pushCodeVerifier } from "./pkce-helper";
+import {
+  clearPersistedData,
+  loadAccessToken,
+  loadUser,
+  persistAccessToken,
+  persistUser,
+  TokenResponse,
+  User,
+} from "./token";
+import { chompSlash, formUrlEncode, hereWithoutSearch } from "./util";
 
 export class Transposit {
-  private claims: ClientClaims | null = null;
+  private hostedAppOrigin: string;
+  private accessToken: string | null;
+  private user: User | null;
+
+  constructor(hostedAppOrigin: string = "") {
+    this.hostedAppOrigin = chompSlash(hostedAppOrigin);
+    this.accessToken = null;
+    this.user = null;
 
-  constructor(private hostedAppOrigin: string = "") {
-    this.claims = this.loadClaims();
+    this.load();
```

Comment:
I agree, it's nice to have it come from a network request because it's a route where the security doesn't rely on the dev ensuring no XSS vulnerability exists.

Reply:
We've decided to load the user via network request. I'll try that out and see how it feels.